app/src/components/Modals/TransactionSendModal.svelte

```diff
@@ -1,43 +1,74 @@
 <script lang="ts" generics="T extends { signature: string }">
+	import Error from '../Statuses/Error.svelte';
+	import Pending from '../Statuses/Pending.svelte';
+	import TransactionSent from '../Statuses/TransactionSent.svelte';
 	import CancelConfirmModal from './CancelConfirmModal.svelte';
 
-	type signature = string;
-
 	export let open: boolean = false;
 	export let sendTransactionHandler: () => Promise<T>;
 
 	let disabled = false;
-	let shouldSendTransaction = false;
 
-	function reset() {
-		disabled = false;
-		shouldSendTransaction = false;
+	enum TxStatus {
+		NotInitialized,
+		Pending,
+		TransactionSent,
+		Error
+	}
+
+	let status = {
+		component: Pending,
+		tx: TxStatus.NotInitialized,
+		statusString: ''
+	};
+
+	let statusComponent = Pending;
+	let statusString: string = '';
+
+	$: if (!open) clearStatus();
```

Comment:
is double clear fine?

Reply:
double? where?

Reply:
on initialization there is additional clear, but that's just svelte